### DISCUSSION TITLE
`use_git_branch = true` does not work when cwd is not git root directory

### DISCUSSION THREAD

**Discussion by mrloop** [olimorris/persisted.nvim#149] [Q&A] [answered]
I have a monorepo, invoking `nvim` from the git root directory and sessions are saved with the git branch in the file name. Invoking `nvim` from a sub directory and sessions are saved without the git branch in the file name. Is this just my setup or are others experiencing this issue as well?

**Comment by olimorris** [comment]
Yes this is the intended design and no plans to add this as functionality. If you browse the init.lua file you can see the plugin's logic.

It may well be possible to add this to your own config by extending the plugin.

**Comment by mrloop** [comment, marked as answer]
My lazy config for future searchers wondering how to use in mono repos  

```lua
{
    "olimorris/persisted.nvim",
    lazy = false, -- make sure the plugin is always loaded at startup
    opts = {
      autoload = true,
      autosave = true,
      use_git_branch = true,
    },
    config = function(_, opts)
      local persisted = require("persisted");
      -- use_git_branch not working from sub directory of git project https://github.com/olimorris/persisted.nvim/discussions/149
      persisted.branch = function()
        local branch = vim.fn.systemlist("git branch --show-current")[1]
        return vim.v.shell_error == 0 and branch or nil
      end
      persisted.setup(opts);
    end,
    init = function()
      vim.o.sessionoptions = "blank,buffers,curdir,folds,help,tabpages,winsize,winpos,terminal,localoptions"
    end
  }
```

**Reply by olimorris** [reply]
This is great! I'll add this to the README and credit you.

**Reply by guoliang** [reply]
nice, was looking for this too
